src/dac.rs

```diff
@@ -1,8 +1,54 @@
 ///! Stabilizer DAC management interface
 ///!
-///! The Stabilizer DAC utilize a DMA channel to generate output updates.  A timer channel is
-///! configured to generate a DMA write into the SPI TXFIFO, which initiates a SPI transfer and
-///! results in DAC update for both channels.
+///! # Design
+///!
+///! Stabilizer DACs are connected to the MCU via a simplex, SPI-compatible interface. Each DAC
+///! accepts a 16-bit output code.
+///!
+///! In order to maximize CPU processing time, the DAC code updates are offloaded to hardware using
+///! a timer compare channel, DMA stream, and the DAC SPI interface.
+///!
+///! The timer comparison channel is configured to generate a DMA request whenever the comparison
+///! occurs. Thus, whenever a comparison happens, a single DAC code can be written to the output. By
+///! configuring a DMA stream for a number of successive DAC codes, hardware can regularly update
+///! the DAC without requiring the CPU.
+///!
+///! In order to ensure alignment between the ADC sample batches and DAC output code batches, a DAC
+///! output batch is always exactly 1 batch after the ADC batch that generated it.
+///!
+///! The DMA transfer for the DAC output codes utilizes a double-buffer mode to avoid losing any
+///! transfer events generated by the timer (for example, when 2 update cycles occur before the DMA
+///! transfer completion is handled). In this mode, there should always be a valid buffer in the
+///! "next-transfer" double-buffer location for the DMA transfer. Once a transfer completes,
+///! software then has exactly one batch duration to properly configure the next buffer before it
+///! begins transfer.
+///!
+///! ## Multiple Samples to Single DAC Codes
+///!
+///! For some applications, it may be desirable to generate a single DAC code from multiple ADC
+///! samples. In order to maintain timing characteristics between ADC samples and DAC code outputs,
+///! applications are required to generate one DAC code for each ADC sample. To accomodate mapping
+///! multiple inputs to a single output, the output code can be repeated a number of times in the
+///! output buffer corresponding with the number of input samples that were used to generate it.
+///!
+///!
+///! # Note
+///!
+///! There is a very small amount of latency between updating the two DACs due to bus matrix
```

Comment:
How much latency? We describe it in terms of "timer ticks" below, but nano- or microseconds would also be nice.

Reply:
This is the same as with the ADCs. It's actually an undeterministic number of AXI bus cycles, hence why it's unspecified.